File: Questions/215. Kth Largest Element in an Array/Solution.cpp
#include<iostream>
using namespace std;

int findKthLargest(vector<int>& nums, int k) {
    auto [min_it, max_it] = minmax_element(nums.begin(), nums.end());
    int minVal = *min_it, maxVal = *max_it;
    int size = maxVal - minVal + 1;
    vector<int> bucket(size, 0);

    for (int num : nums) {
        bucket[num - minVal]++;
    }

    for (int i = size - 1; i >= 0; i--) {
        k -= bucket[i];
        if (k <= 0) return i + minVal;
    }
    return 0;

}

int main() {
    vector<int> nums = {2, 1, 5, 6, 4};
    int ret = findKthLargest(nums, 2);
    cout << ret << endl;
    return 0;
}

File: Questions/215. Kth Largest Element in an Array/Solution.java
// Input: nums = [3,2,1,5,6,4], k = 2
// Output: 5

import java.util.*;

public class Solution {
    public int findKthLargest(int[] nums, int k) {
        int maxValue = Integer.MIN_VALUE, minValue = Integer.MAX_VALUE;
        for (int num : nums) {
            maxValue = Math.max(maxValue, num);
            minValue = Math.min(minValue, num);
        }
        int size = maxValue - minValue + 1;
        int[] bucket = new int[size];
        Arrays.fill(bucket, 0);

        for (int num : nums) {
            bucket[num - minValue]++;
        }

        for (int i = size - 1; i >= 0; i--) {
            k -= bucket[i];
            if (k <= 0) {
                return i + minValue;
            }
        }
        return -1;
    }

}
